uaclient/cli.py

```diff
@@ -477,6 +562,127 @@ def get_valid_entitlement_names(names: "List[str]"):
     return entitlements_found, entitlements_not_found
 
 
+def action_config(args, cfg, **kwargs):
+    """Perform the config action.
+
+    @return: 0 on success, 1 otherwise
+    """
+    parser = get_parser()
+    subparser = parser._get_positional_actions()[0].choices["config"]
+    valid_choices = subparser._get_positional_actions()[0].choices.keys()
+    if args.command not in valid_choices:
+        parser._get_positional_actions()[0].choices["config"].print_help()
+        raise exceptions.UserFacingError(
+            "\n<command> must be one of: {}".format(", ".join(valid_choices))
+        )
+
+
+def action_config_show(args, cfg, **kwargs):
+    """Perform the 'config show' action.
+
+    @return: 0 on success, 1 otherwise
+    """
+    col_width = str(max([len(x) for x in config.UA_CONFIGURABLE_KEYS]) + 1)
+    row_tmpl = "{key: <" + col_width + "}: {value}"
+    for key in config.UA_CONFIGURABLE_KEYS:
+        print(row_tmpl.format(key=key, value=getattr(cfg, key)))
+
+
+@assert_root
+def action_config_set(args, cfg, **kwargs):
+    """Perform the 'config set' action.
+
+    @return: 0 on success, 1 otherwise
+    """
+    from uaclient.apt import setup_apt_proxy
+    from uaclient.entitlements.livepatch import (
+        configure_livepatch_proxy,
+        configure_snap_proxy,
+    )
+
+    parser = get_parser()
+    config_parser = parser._get_positional_actions()[0].choices["config"]
+    subparser = config_parser._get_positional_actions()[0].choices["set"]
```

Comment:
driveby thought that we shouldn't address now. If we need access to the parser/subparser often in our action functions, maybe we should pass them as args

Reply:
This is a good suggestion, I toyed with this thought too and wondered about a slight refactor for that. This approach I took  is definitely bad practice as I'm trying to reference private methods on the parser which I assume could change without warning in the future and break us.

Reply:
We could instead store the valid choices in a static variable. This will avoid the need of using a private method from argparser to discover that.

Reply:
The confusing motivation of this code is actually two fold (in differnt functions):
  1. to check valid keys() which we could easily persist somewhere as a static var or lookup function
  2. to call that subparser.print_help().   
  
Maybe for case 2 we actually   collect a global of arg_subparsers keyed on the subcommand "ua.config.set" and then we don't have to look it up using private methods. Again this is probably a separate PR, but we can discuss depending on what @lucasmoura and @orndorffgrant think.

Reply:
I created a Jira issue for us to come back to this here: https://warthogs.atlassian.net/browse/SC-205

Resolving this now